### PR TITLE
Obsolete auto_inject_python_workers flag and make it default

### PR DESCRIPTION
**Pull request by ryanking13**
I tested the compat flag locally, so removing the compat flag and turning it on by default. After merging this, we need to update wrangler and pywrangler too to make this available to normal users.

### CODE REVIEW

File: src/workerd/io/compatibility-date.capnp [hunk 1]
```diff
@@ -1664,11 +1664,9 @@ struct CompatibilityFlags @0x8f8c1b68151b6cef {
...
 
-  autoInjectPythonWorkers @189 :Bool
+  obsolete189 @189 :Bool
       $compatEnableFlag("auto_inject_python_workers")
       $compatDisableFlag("no_auto_inject_python_workers")
       $experimental;
-  # When enabled, a Worker whose entrypoint is Python are automatically
-  # considered as a Python Worker. This flag will be obsoleted once the feature
-  # is stable.
+  # Obsolete flag. Has no effect. Still accepted so configs which set it keep validating.
```
[the remark this answers is not in the document]
Comment (ryanking13) [reply src/workerd/io/compatibility-date.capnp:1671]: This flag was experimental so there is no user worker that sets this flag. So this is fine to obsolete

**Comment by ryanking13** [comment]
@ask-bonk try again

**Review by dom96 — APPROVED** [review]
Looks good, though I wonder if we should inject this flag into all of our tests (including E2E) first to make sure there aren't any weird edge cases.

**Comment by ryanking13** [comment]
> Looks good, though I wonder if we should inject this flag into all of our tests (including E2E) first to make sure there aren't any weird edge cases.

Yeah, probably. But it would require updating the python_workers check from `workers_sdk` and `pywrangler` which is somewhat annoying order-wise. Alternatively, I think I can try removing python_wokers flags from workerd unittests to see if there are some weird edge cases.

**Review by hoodmane — APPROVED** [review]

**Comment by hoodmane** [comment]
Thanks!
